service/storage_service.cc

```diff
@@ -391,7 +391,7 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
     co_await db::schema_tables::recalculate_schema_version(_sys_ks, proxy, _feature_service);
 
     assert(_group0);
-    co_await _group0->setup_group0(_sys_ks.local());
+    co_await _group0->setup_group0(_sys_ks.local(), initial_contact_nodes);
```

Comment:
This changes the semantics of discovery. initial_contact_nodes != gossiper.get_seeds(). There must be a good reason to not use gossiper.get_seeds() here. I'd avoid more coupling of the old bootstrap and the new bootstrap.

Reply:
During bootstrap, `initial_contact_nodes == gossiper.get_seeds()`. Later, when we restart and we already know some peers, we use them. But at this point `setup_group0` is just restarting an existing server.